flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/window/tvf/common/WindowAggOperator.java

```diff
@@ -222,7 +225,10 @@ public void processElement(StreamRecord<RowData> element) throws Exception {
     @Override
     public void processWatermark(Watermark mark) throws Exception {
         if (mark.getTimestamp() > currentWatermark) {
-            windowProcessor.advanceProgress(mark.getTimestamp());
+            // advance the window processor by timer if this window agg is based on proctime
```

Comment:
-> 'If this is a proctime window, progress should not be advanced by watermark, or it'll disturb timer-based processing'?